File: basicProgramming/dart/ifElse/p6.dart
import 'dart:io';

void main() {
  stdout.write('Enter any character: ');

  var ch = String.fromCharCode(stdin.readLineSync()!.codeUnitAt(0));

  List<String> vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
  bool isVowel = false;

  for (int i = 0; i < vowels.length; i++) {
    if (ch == vowels[i]) {
      isVowel = true;
      break;
    }
  }

  if (isVowel) {
    print('$ch is a vowel.');
  } else {
    print('$ch is a consonant.');
  }
}

File: basicProgramming/c++/ifEsle/p6.cpp
#include <iostream>
using namespace std;

int main()
{
    char ch;

    cout << "Enter any character: ";
    cin >> ch;

    char vowels[] = {
        'a',
        'e',
        'i',
        'o',
        'u',
        'A',
        'E',
        'I',
        'O',
        'U',
    };

    int size = sizeof(vowels) / sizeof(vowels[0]);
    bool isVowel = false;

    for (int i = 0; i < size; i++)
    {
        if (ch == vowels[i])
        {
            isVowel = true;
            break;
        }
    }

    if (isVowel)
    {
        cout << ch << " is a vowel." << endl;
    }
    else
    {
        cout << ch << " is a consonant." << endl;
    }

    return 0;
}
